server/src/main/java/io/spine/server/command/Command.java

```diff
@@ -0,0 +1,88 @@
+import java.lang.annotation.Target;
+
+/**
+ * Marks a commanding method.
+ *
+ * <p>A commanding method <strong>must</strong>:
+ * <ul>
+ *     <li>be annotated with {@link Command @Command};
+ *     <li>return a command message derived from {@link com.google.protobuf.Message Message},
+ *     if there is only one command to be generated;
+ *     <strong>or</strong> an {@link java.lang.Iterable Iterable} of command messages for two or
+ *     more commands;
+ *     <li>accept one of the following as the first parameter:
+ *     <ul>
+ *        <li>a {@linkplain com.google.protobuf.Message command message}
+ *        <li>an {@linkplain com.google.protobuf.Message event message}
+ *        <li>a {@link com.google.protobuf.Message rejection message}
+ *     </ul>
+ * </ul>
+ *
+ * <p>A commanding method <strong>may</strong> accept a context of the incoming message
+ * ({@link io.spine.core.CommandContext CommandContext},
+ * {@link io.spine.core.EventContext EventContext}, or
+ * {@link io.spine.core.RejectionContext RejectionContext} correspondingly)
+ * as the second parameter, if handling of the command requires its context.
+ *
+ * <p>If the annotation is applied to a method which doesn't satisfy any of these requirements,
+ * this method is not considered a command handler and is <strong>not</strong> registered for
+ * command generation.
+ *
+ * <h2>Throwables</h2>
+ * <p>A commanding method may throw a {@link io.spine.base.ThrowableMessage ThrowableMessage}
+ * <em>if, and only if</em> it accepts a command message. This means that the incoming command
+ * may be either rejected, or substituting command(s) must be generated.
+ *
+ * <p>Throwing {@linkplain Throwable other types} in command transforming methods in not allowed.
+ *
+ * <p>Commanding methods accepting events or rejections may not throw.
+ *
+ * <h2>Access Modifiers</h2>
+ *
+ * <p>A commanding method <strong>should</strong> have package-private access. It will allow
+ * calling this method from tests. The method should not be {@code public} because it is not
+ * supposed to be called directly.
+ *
+ * <h2>Handle or Transform a Command</h2>
+ *
+ * <p>An application can either {@linkplain io.spine.server.command.Assign handle} an incoming
+ * command <strong>or</strong> transform it into one or more commands that will be handled instead.
+ * Because of this, a commanding method cannot accept a command message, if there is already a
+ * command handling method which accepts the command message type.
+ *
+ * <p>If a commanding method accepts a command type which is also handled by the application
+ * by another method, a run-time error will occur. This also means that an application cannot have
+ * two commanding methods that accept the same command type.
+ *
+ * @author Alexander Yevsyukov
+ * @see io.spine.server.command.Assign Handling Commands
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface Command {
```

Comment:
This name is confusing. I bet there will be a file in the `server` module which uses both `io.spine.server.command.Command` and `io.spine.core.Command`. I do understand that one of these is a verb and another is a noun. Though, my understanding does not make the situation better.

Can we rename this to something like `@Transform`?

Reply:
Yes, it's a verb. There will be other methods that issue command messages in response to events or rejections. It is better to have one annotation for all the cases.

The client code is not going to see the `core.Command` at all.